File: src/main/java/graphgenerator/JsonFileGenerator.java
package graphgenerator;

import DataConstuctor.Daily;
import org.json4s.JsonAST;
import scala.Tuple2;
import scala.collection.immutable.List;

import java.io.OutputStream;
import java.nio.file.Path;
import java.util.Arrays;
import java.util.Collections;

public class JsonFileGenerator {

    private String jsonFileName;

    public JsonFileGenerator(String jsonFileName) {
        this.jsonFileName = jsonFileName;
    }

    public void generateJson(Daily[] dailies, Path outputDirectory) {
        OutputStream jsonFileOutputStream = getJsonFileOutputStream(outputDirectory);
        JsonAST.JValue dailiesJson = buildDailiesJson(dailies);
    }

    private OutputStream getJsonFileOutputStream(Path outputDirectory) {
        return null;
    }

    private JsonAST.JValue buildDailiesJson(Daily[] dailies) {
        JsonAST.JSet dailyValues = Arrays.stream(dailies).map(daily -> {

                }).collect();
        return new JsonAST.JObject(
                (List<Tuple2<String,JsonAST.JValue>>) Collections.singletonList(
                        new Tuple2<String, JsonAST.JValue>("dailies", dailyValues)));
    }


}

File: src/main/scala/graphgenerator/JsonFileGenerator.scala
package graphgenerator

import java.io.{FileOutputStream, OutputStream}
import java.nio.file.Path

import DataConstuctor.Daily
import org.json4s.JsonAST._
import org.json4s.JsonDSL._
import org.json4s.jackson.JsonMethods._

class JsonFileGenerator(jsonFileName: String) {

  def generateJson(dailies: Array[Daily], totalHostnames: Int, outputDirectory: Path): Unit = {
    val outputStream: OutputStream = getOutputStream(outputDirectory)
    outputStream.write(pretty(render(buildJson(dailies, totalHostnames))).getBytes)
    outputStream.flush()
    outputStream.close()
  }

  private def getOutputStream(outputDirectory: Path) = {
    new FileOutputStream(outputDirectory.resolve(jsonFileName).toFile)
  }

  private def buildJson(dailies: Array[Daily], totalHostnames: Int): JValue = {
    ("totalHostnames" -> totalHostnames) ~
    ("dailies" -> JSet(dailies.toStream.map((daily: Daily) => {
      ("todayUniqueHostnames" -> daily.todayUniqueHostnames) ~ ("totalUniqueHostnames" -> daily.totalUniqueHostnames)
        }).toSet))
  }

}
